File: dotnet/Lexy.Poc/Lexy.Poc.Core/Parser/Tokens/CommentToken.cs

namespace Lexy.Poc.Core.Parser.Tokens
{
    public class CommentToken : ParsableToken
    {
        public CommentToken(TokenCharacter character) : base(character)
        {
        }

        public override ParseTokenResult Parse(TokenCharacter character, IParserContext parserContext)
        {
            AppendValue(character.Value);

            return ParseTokenResult.InProgress();
        }

        public override ParseTokenResult Finalize(IParserContext parserContext)
        {
            return ParseTokenResult.Finished(true);
        }
    }
}

File: typescipt/src/parser/tokens/commentToken.ts
import {ParsableToken} from "./parsableToken";
import {TokenCharacter} from "./tokenCharacter";
import {newParseTokenFinishedResult, newParseTokenInProgressResult, ParseTokenResult} from "./parseTokenResult";
import {TokenType} from "./tokenType";

export function instanceOfCommentToken(object: any): boolean {
  return object?.tokenType == TokenType.CommentToken;
}

export function asCommentToken(object: any): CommentToken | null {
  return instanceOfCommentToken(object) ? object as CommentToken : null;
}

export class CommentToken extends ParsableToken {

  public tokenIsLiteral: boolean = false;
  public tokenType = TokenType.CommentToken;

  constructor(character: TokenCharacter) {
    super(character);
  }

  public parse(character: TokenCharacter): ParseTokenResult {
    this.appendValue(character.value);
    return newParseTokenInProgressResult();
  }

  public finalize(): ParseTokenResult {
    return newParseTokenFinishedResult(true);
  }
}